wrappers/services.go

```diff
@@ -516,13 +520,20 @@ func EnsureSnapServices(snaps map[*snap.Info]*SnapServiceOptions, opts *EnsureSn
 		}
 	}()
 
-	handleFileModification := func(s *snap.Info, app *snap.AppInfo, path string, content []byte) error {
+	handleFileModification := func(s *snap.Info, app *snap.AppInfo, unitType string, path string, content []byte) error {
 		old, modifiedFile, err := tryFileUpdate(path, content)
 		if err != nil {
 			return err
 		}
 
 		if modifiedFile {
+			if observeChange != nil {
```

Comment:
I was thinking about how to best implement this, I wonder if it's not more "logical" to instead call the callback at the end of the function instead after we know that all of the modifications were successful otherwise we may have situations where the callback is called for successful modifications and does some side-effect thing that we wouldn't want to have happen if the modifications are rolled back.

then again, we don't have a use case for this at the moment, so maybe it's fine to leave as-is or change later if we need more complicated callbacks. It would also complicate this code a bit I think but probably not unreasonably so

Reply:
That's an interesting remark, but I agree it's not super important until we have a case for it.